src/cmdc_tools/datasets/official/OK/counties/tulsa.py

```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+import textwrap
+
+from ... import ArcGIS
+from .... import DatasetBaseNoDate
+
+
+class OKTulsa(DatasetBaseNoDate, ArcGIS):
+    """
+    Tulsa, Oklahoma publishes their county level data in a dashboard that can
+    be found at:
+
+    https://tcchd.maps.arcgis.com/apps/opsdashboard/index.html#/ebb119cd215b4c57933b7fbe477e7c30
+
+    This retrieves the data from the ArcGIS API
+    """
+
+    ARCGIS_ID = "yHSoWow4TMdKqS8y"
+    source = (
+        "https://tcchd.maps.arcgis.com/apps/opsdashboard/index.html#/"
+        "ebb119cd215b4c57933b7fbe477e7c30"
+    )
+    state_fips = 40
+
+    def __init__(self, params=None):
+        super(OKTulsa, self).__init__(params=params)
+
+        return None
+
+    def _insert_query(self, df, table_name, temp_name, pk):
+        out = f"""
+        INSERT INTO data.{table_name} (vintage, dt, fips, variable_id, value)
+        SELECT tt.vintage, tt.dt, us.fips, mv.id as variable_id, tt.value
+        FROM {temp_name} tt
+        INNER JOIN meta.us_fips us ON tt.county=us.name
+        LEFT JOIN meta.covid_variables mv ON tt.variable_name=mv.name
+        WHERE us.state = LPAD({self.state_fips}::TEXT, 2, '0')
```

Comment:
I feel like I've seen this exact insert query before. Is this generic enough that we could refactor a bit do DRY?

Reply:
This query shows up a few places (and, if I'm being honest, I basically copied and pasted it which is embarrassing to me).

Refactoring it has been on my TODO list but I've kept putting it off. I think we would basically just need to have a child-class of `DatasetBaseNoDate` called `DatasetBaseNoDateNoFips` (or something like that) where it relied on joining by state/county name...

Actually writing this has persuaded me into doing it in this PR... I'll ping you when it's ready again.

Reply:
Refactored.